### DISCUSSION TITLE
rel="me" link in profile

### DISCUSSION THREAD

**Discussion by satchlj** [community/community#6248] [Other Feature Feedback, Questions, & Ideas]
It would be nice to have the rel attribute in the source code of the README.md file in my profile repo be preserved when it is rendered. See how it is rendered in html:
````
<div id="readme"  class="Box-body readme blob js-code-block-container p-5 p-xl-6 gist-border-0">
<article class="markdown-body entry-content container-lg" itemprop="text"><p><g-emoji class="g-emoji" alias="wave" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f44b.png">👋</g-emoji> Hi, I’m Satya Johnson
 I'm slowly learning how to build websites for real (not using a CMS).
 I have a VPS running debian and it is my playground!<br>
  <a href="https://satchlj.com/about/" rel="nofollow">About Me</a><br>
  <a href="https://tilde.zone/@satchlj" rel="nofollow">Mastodon</a></p>
  </article>
  </div>
````
compared to the source code in README.md:
````
👋 Hi, I’m Satya Johnson
I'm slowly learning how to build websites for real (not using a CMS).
I have a VPS running debian and it is my playground!  
[About Me](https://satchlj.com/about/)  
<a rel="me" href="https://tilde.zone/@satchlj">Mastodon</a>
````

**Comment by weiji14** [comment]
Referencing https://jan.wildeboer.net/2022/06/Quick-Fix-Verified-Mastodon-Github and https://indieweb.org/rel-me#Mastodon, try going to your GitHub profile page, click 'Edit Profile' on the left, and put your Mastodon profile link (e.g. `https://mastodon.social/@username`) as your website link :link:, that should result in some HTML on your profile page like:

```
<a rel="nofollow me" class="Link--primary" href="https://mastodon.social/@username">https://mastodon.social/@username</a>
```
Now go back to your Mastodon settings, add your GitHub profile link (e.g. `https://github.com/username`), and you should get the green verification :grin:

Pro tip: If you don't see the green verification immediately, try adding a slash `/` to your GitHub URL (i.e. `https://github.com/username/`), save, and refresh the page. If it still doesn't work, just wait a while and give the cache a chance to update.

**Reply by magnus919** [reply]
Sadly this only works for those who are happy to list Mastodon as their one and only web site.

GitHub only seems to provide very few social fields. It's a shame. It'd be nice to have links back to our LinkedIn, Mastodon, etc. on the side panel the the "rel=me" metadata included.

Barring that, being able to accomplish same in Markdown would be ideal.

**Reply by 3894419350** [reply]
Jggkkkf

**Reply by boozook** [reply]
Thanks! But what about same for project-repo, not for personal- or org- profile?

**Comment by howbazaar** [comment]
Alternatively, have a dedicated "fediverse" field in the profile like the "twitter" one

**Reply by tom-sherman** [reply]
IMO this should only be the solution if there are some security or technical concerns that make `rel="me"` in markdown not viable.

**Reply by ashwinvis** [reply]
I agree. The correct solution is to whitelist rel="me" and not sanitize it.

**Reply by joshuapsteele** [reply]
How do you do this? I'm struggling to understand exactly what the suggested solution is here.

**Reply by tom-sherman** [reply]
@jsteelepfpt Workaround solution here: https://github.com/orgs/community/discussions/6248#discussioncomment-4061508

Essentially put your mastodon profile URL in the "Website" field on your profile.

@howbazaar Suggested adding a new "fediverse" field which would live alongside the "Website" and "Twitter" fields on your profile.

The OP in this discussion suggested allowing you to create `<a rel="me">` links in markdown, this would allow you to add it to your `username/username/README.md` file to verify your profile URL. In my opinion this is the preferred solution for GitHub to implement.

**Reply by thestinger** [reply]
Mastodon currently doesn't revalidate URLs so you can temporarily add it to your profile and then remove it afterwards. It will likely eventually have support for revalidating them since the status quo is a security issue with the verified link support due to being able to keep links verified after you no longer control them.

**Comment by orta** [comment]
This request feels a bit more pressing over the last few weeks as twitter continues to become less of a useful platform for programming communities, and an unreliable system for identity

**Comment by thestinger** [comment]
It would also be nice if rel="me" was added to the URL for organizations. It's already done for users but not organizations. For example, the URL at https://github.com/GrapheneOS linking to https://grapheneos.org/ should have rel="me". After all, the whole purpose of that URL is providing a link to a website representing the organization.

**Comment by CaptainOfPhB** [comment]
Let me tell you guys how to do it 😂

DON'T edit your GitHub profile `README.md` file, that will no work.

Fill the link section of your GitHub profile page with your mastodon link, then it will take effect, my English is poor so see the below screenshot.

<img width="636" alt="image" src="https://user-images.githubusercontent.com/30765485/204231657-d50ba1c2-c63b-49e2-ab7d-a2e9df7e676a.png">

After a while my mastodon profile page verified it's my website.

![mastodon world_@captainofphb](https://user-images.githubusercontent.com/30765485/204232958-322cc8cd-ed22-4a28-8289-5c707739d8d7.png)

You can then undo your edits or leave the link section blank. Verified status will be preserved.

Wish it can help you.

**Comment by ashwinvis** [comment]
Actually some of us would like to have the website in that field. And rel=me is not unique to Mastodon. It is useful to also use with IndieAuth

https://indieweb.org/IndieAuth

Therefore it would be nice to have it supported elsewhere.
-------- Original Message --------

**Reply by CaptainOfPhB** [reply]
Sure, you are totally right, it's just a hack trick for the mastodon case 😂

**Reply by zemlanin** [reply]
Aren’t other servers verify links independently? So while your instance and other currently following you would load GitHub `rel="me"`, new followers won’t see the checkmark after you’ll remove it from the profile?..

**Reply by ashwinvis** [reply]
Good idea. I should test that.

**Comment by ashwinvis** [comment]
It seems that Mastodon queries the rel=me link only once and caches the verification. So here is what I did now

1. Add Mastodon to profile in Github
2. Complete Mastodon Github verification
3. Remove Mastodon from Github and add back my website.

It is a hack and I don't know if this needs to be repeated later.

**Reply by numist** [reply]
The cache expires, so this isn't a reliable method.

**Comment by derekprior** [comment]
Hey folks, I just wanted to give you a heads up that we are working on this. It'll ship when it's ready, but with holiday season vacations coming up that won't be for a bit yet. Thanks for your patience.

**Comment by therealchjones** [comment]
> Hey folks, I just wanted to give you a heads up that we are working on this. It'll ship when it's ready, but with holiday season vacations coming up that won't be for a bit yet. Thanks for your patience.

@derekprior Thanks so much for the update and for your work on this. Is there a discussion, issue, or other place we can follow along the (or contribute to) the points being considered?

**Reply by derekprior** [reply]
We don't have any public issue for discussion, but you can keep an eye on our [changelog](https://github.blog/changelog/) or, perhaps ironically, [Twitter](https://twitter.com/github) for notice when it's available

**Reply by shengslogar** [reply]
Kudos for the human responses here 🥂

**Comment by keithamus** [comment]
I believe this is resolved, right @derekprior ?

**Reply by derekprior** [reply]
Correct. This feature has shipped: https://github.blog/changelog/2023-02-02-add-more-social-links-to-your-user-profile/

**Reply by rdela** [reply]
@derekprior this discussion is about `rel="me"` being overwritten by `rel="nofollow"` even if one puts `rel="me nofollow"` in the markup. [This commit comment](https://github.com/rdela/rdela/commit/295a1b718639a026e46f8c555326bb7ea6af0128#commitcomment-108112897) is illustrative. That part is not resolved.

**Reply by rdela** [reply]
However, when you add your Mastodon profile in social links, `rel="nofollow me"` does get added to that link so one can add their GH profile and get the link verified on Mastodon, which was kinda the point of the discussion.

**Reply by satchlj** [reply]
Yeah, it’s kind of hilarious to hear “we are working on this” and then later discover that my entire point was missed. @rdela thanks for pointing this out.
